### ISSUE TITLE
Feature idea: callbacks should also receive the model output(s) for the current batch

### ISSUE THREAD

**Issue by isarandi** [fchollet/keras#6639]
Currently `Callback.on_batch_end` only gets passed the current batch index and the loss values. If we also had access to the model's output(s) we could perform some more powerful custom actions here, e.g. if the network's output is an image (such as a segmentation mask etc.) we could visualize it. If the output is a set of bounding box coords, we could visualize that.

Seeing what the network currently does is very valuable for debugging and inspection. Currently this can only be done using an explicit TensorFlow training loop if I'm not mistaken.

**Comment by kgrm** [comment]
You can get outputs from callbacks using `self.model.predict`, but this requires re-running data processing and prediction for no reason, so I agree this would be a useful addition.
